exporter/collector/logs_test.go

```diff
@@ -210,6 +218,32 @@ func TestLogMapping(t *testing.T) {
 			},
 			maxEntrySize: defaultMaxEntrySize,
 		},
+		{
+			name: "log body with string value and error converted to json payload",
+			mr: func() *monitoredrespb.MonitoredResource {
+				return nil
+			},
+			log: func() plog.LogRecord {
+				log := plog.NewLogRecord()
+				log.SetSeverityNumber(18)
+				log.Body().SetStr("{\"message\": \"hello!\"}")
+				return log
+			},
+			expectedEntries: []logging.Entry{
+				{
+					Payload: map[string]interface{}{
+						"@type":   "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
+						"message": `{"message": "hello!"}`,
```

Comment:
note that "json" bodies (that are actually string values) can end up with a redundant message key. I don't think we should try automatically parsing bodies that are string valued to JSON and making assumptions to remove them